### PR TITLE
Bond/react delete atoms

### PR DESCRIPTION
**Pull request by jrgissing**
## Purpose

allows deletion of a user-specified set of atoms, such as reaction by-products or other molecules.
thanks to several users for requesting this feature

## Author(s)

JG

## Backward Compatibility

yes

## Implementation Notes

_Provide any relevant details about how the changes are implemented, how correctness was verified, how other features - if any - in LAMMPS are affected_

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [ ] One or more example input decks are included
- [x] The source code follows the LAMMPS formatting guidelines

## Further Information, Files, and Links

_Put any additional information here, attach relevant text or image files, and URLs to external sites (e.g. DOIs or webpages)_

### CODE REVIEW

File: doc/src/fix_bond_react.txt [hunk 1]
```diff
@@ -79,7 +80,7 @@ probabilistic criteria to effect predetermined topology changes in
 simulations using standard force fields.
 
 This fix was created to facilitate the dynamic creation of polymeric,
-amorphous or highly cross-linked systems. A suggested workflow for
+amorphous or highly-crosslinked systems. A suggested workflow for
```
Comment (akohlmey) [inline doc/src/fix_bond_react.txt]: to the best of my knowledge "highly cross-linked" is the correct grammar

File: doc/src/fix_bond_react.txt [hunk 2]
```diff
@@ -70,7 +70,8 @@ changes will be referred to as 'reactions' throughout this
 documentation. Topology changes are defined in pre- and post-reaction
 molecule templates and can include creation and deletion of bonds,
 angles, dihedrals, impropers, bond-types, angle-types, dihedral-types,
```
Comment (akohlmey) [inline doc/src/fix_bond_react.txt]: this should be "atom types", "bond types" and so on. the only use of these term with a hyphen is in the command summary, where the hyphen signifies, that the two words stand for a single parameter.

File: doc/src/fix_bond_react.txt [hunk 3]
```diff
@@ -309,7 +313,16 @@ edge are unaffected by this setting.
 
 A few other considerations:
 
-It may be beneficial to ensure reacting atoms are at a certain
+Many reactions result in one or more atoms that are considered
+unwanted by-products. Therefore, bond/react provides the option to
+delete a user-specified set of atoms. These pre-reaction atoms are
+identified in the map file. A deleted atom must still be included in
+the post-reaction molecule template, in which it cannot be bonded to
+an undeleted atom. In addition to deleting unwanted reaction
```
Comment (akohlmey) [inline doc/src/fix_bond_react.txt]: `undeleted` also shows up in the spellcheck part of the documentation build test. however, my impression is, that `undeleted` (i.e. previously deleted and now revived) may not be the correct term, but `not deleted` or `remaining` or something similar. can you please clarify? if undelete is the correct term, please also add to the list of false positives.

File: doc/src/fix_bond_react.txt [hunk 4]
```diff
@@ -203,15 +204,16 @@ A discussion of correctly handling this is also provided on the
...
 A map file has a header and a body. The header of map file the
-contains one mandatory keyword and two optional keywords. The
+contains one mandatory keyword and three optional keywords. The
 mandatory keyword is 'equivalences' and the optional keywords are
-'edgeIDs' and 'customIDs':
+'edgeIDs' and 'deleteIDs' and 'customIDs':
 
 N {equivalences} = # of atoms N in the reaction molecule templates
 N {edgeIDs} = # of edge atoms N in the pre-reacted molecule template
+N {deleteIDs} = # of atoms N that are specified for deletion
```
Comment (akohlmey) [inline doc/src/fix_bond_react.txt]: Please add `deleteIDs` to `doc/utils/sphinx-config/false_positives.txt`

File: src/USER-MISC/fix_bond_react.cpp [hunk 5]
```diff
@@ -1650,6 +1653,18 @@ void FixBondReact::find_landlocked_atoms(int myrxn)
     }
   }
 
+  // additionally, if a deleted atom is bonded to a undeleted atom, bad
```
Comment (akohlmey) [inline src/USER-MISC/fix_bond_react.cpp]: see my previous inquiry about what is meant by `undeleted` in this context. if it needs to be changed in the docs, it should be changed here, too.

File: doc/src/fix_bond_react.txt (hunk 3 above)
Comment (jrgissing) [reply doc/src/fix_bond_react.txt]: yes, you're right, thanks

**Pull request merged**